### ISSUE TITLE
amp-carousel : images load intermittently

### ISSUE THREAD

**Issue by anuragphadke** [ampproject/amphtml#27036]
## What's the issue?

I am using amp-carousel-0.2.js, however, images in the <amp-carousel> tag fail to load 1/5 attempts.

## How do we reproduce the issue?

1. Visit: https://beta.webfast.co/amp/image-carousel.html
2. Press cmd-shift-r (hard refresh) and you will notice that the image in the carousel only loads intermittently.

Another way to produce, start Chrome on iOS in incognito mode:
1. Visit: https://beta.webfast.co/amp/image-carousel.html, images don't load at first try.
2. reload the page.
3. images will now load.
The above is repeatable everytime u close all incognito windows on Chrome on iOS.

## What browsers are affected?

All browsers on mobile.

## Which AMP version is affected?
AMP version: 2002192257490
AMP carousel: https://cdn.ampproject.org/v0/amp-carousel-0.2.js

Is this a new issue? Or was it always broken? Paste the version of AMP where you saw this issue. 
Experiencing for first time related to the above page.

**Comment by anuragphadke** [comment]
@alanorozco : any possible workarounds that I can use until this gets fixed?

**Comment by rsmith4321** [comment]
I seem to be having the exact same issue. It seems to be a lazy loading issue the same as described. https://www.ryansmithphotography.com at the top of the page. I don’t notice the issue on iOS.

**Comment by rsmith4321** [comment]
I'm done some more testing on this issue. It seems to be related to the amp-carousel that is generated by Jetpack. Can someone do some testing with amp-carousel v2 and the jetpack gallery?  I've switched to using the code generated from a standard wordpress gallery and just adding the loop and autoplay attributes to get the same functionality as the Jetpack gallery.  Adding the code manually I no longer get missing images in the slideshow.

**Comment by westonruter** [comment]
I can reliably regularly (if yet sporadically) reproduce this issue: https://www.ryansmithphotography.com/

Here's a screen capture showing the problem after several reload attempts: https://youtu.be/nBtjBtai8s0

This may be a similar issue to #26952. 

@jridgewell @choumx Is this Resources bug? Will it be fixed by IntersectionObserver?

**Comment by rsmith4321** [comment]
Thanks for checking this issue.  Like I mentioned I've never seen this
issue when adding the amp-carousel in directly with code in an html block.
So I don't know if there is something different with the way the block is
loaded with jetpack.

**Comment by jridgewell** [comment]
Hi @rsmith4321, thanks for the report!

After investigating, I'm convinced this is caused by a bug in newly launched experiment. The bug fix was merged in https://github.com/ampproject/amphtml/pull/30188, and will rollout to production tomorrow.

**Comment by rsmith4321** [comment]
I'm still able to recreate the issue by doing an empty cache reload. I've also noticed this issue for many months when I use the jetpack carousel so I don't think it's anything recent. I've never seen the issue adding an amp-carousel in an html block.  Thanks.
https://drive.google.com/file/d/1Fr7InOAbm5Ut_BLqZhLcEfpNB25ql7mg/view?usp=sharing

**Comment by jridgewell** [comment]
Wow, this is strange. The amp-img failed to upgrade to the [`AmpImg`](https://github.com/ampproject/amphtml/blob/15bebe67d26b63fa35d796495dce77bcb8367bba/builtins/amp-img.js#L47) `BaseElement`. We're still using the [`ElementStub`](https://github.com/ampproject/amphtml/blob/15bebe67d26b63fa35d796495dce77bcb8367bba/src/element-stub.js#L23). In fact, it's already "laid out":

<img width="1552" alt="Screen Shot 2020-09-28 at 4 56 22 PM" src="https://user-images.githubusercontent.com/112982/94485244-a12c4300-01ab-11eb-9e4d-339950d23fdb.png">

Reopening, setting to a higher priority.

**Comment by rsmith4321** [comment]
Is there is way to know when this update will live? Do we have to wait for a version 0.3 of the amp-carousel?  Thanks!

**Comment by jridgewell** [comment]
It will be live in beta (1% of users) tomorrow afternoon, and production (100%) next Tuesday.
